### ISSUE TITLE
Show/Hide timeline items according to timeline "zoom" level

### ISSUE THREAD

**Issue by mojoaxel** [almende/vis#2586]
It would be nice to be able to define something like a zoom level per timeline item. Tese items than only get displayed, if you are zoomed-in at a given "level". 
The zoom level in witch the item is visible could be defined by something like a `visibleRange`:

```js
item: {
  id: 4, 
  content: 'item', 
  start: '2014-04-16', 
  end: '2014-04-19',
  visibleRange: {
    min: '1 month',
    max: '1 week'
  }
```

So if the zoom range is 1 month or higher (less days), than this item gets displayed. If the user zooms in until the zoom-range gets higher than 1 week (multiple days) than the given item is hidden again.

**Comment by BelalAmin** [comment]
Yes, that would be a briliant feature, if you managed to do it, please tell me how

**Comment by tomdcc** [comment]
We would definitely use this, we're currently looking at how to do this manually.

One comment - for our needs the zoom level would need to be customisable somehow, either by specifying a window size (current end - start or percentage of total) or providing a custom function. Where we need these to kick in isn't on standard date etc blocks.

**Comment by bradh** [comment]
Hiding it according to some user-supplied function might be more general, although perhaps somewhat slower.

**Comment by avrahamcool** [comment]
can this be implemented using a `dataview` with a custom filter?
anyone here found a suitable way for solving this?
